### PR TITLE
ARSN-184 type check s3routes

### PR DESCRIPTION
**Pull request by ghivert**

### CODE REVIEW

**Review by jbertran — APPROVED** [review]

File: lib/s3routes/routes/routePUT.ts [hunk 1]
```diff
@@ -200,14 +212,13 @@ function routePUT(request, response, api, log, statsClient) {
...
-            if (Number.isNaN(request.parsedContentLength) ||
-            request.parsedContentLength < 0) {
+            if (Number.isNaN(parsedContentLength) || parsedContentLength < 0) {
                 return routesUtils.responseNoBody(errors.BadRequest,
                     null, response, 400, log);
             }
-            log.end().addDefaultFields({
-                contentLength: request.parsedContentLength,
-            });
+            // TODO What's happening?
```
Comment (nicolas2bert) [inline lib/s3routes/routes/routePUT.ts]: What is the issue here? Do we have a ticket number related to this that we can add to the comments?

File: lib/s3routes/routesUtils.ts [hunk 2]
```diff
@@ -0,0 +1,1149 @@
...
+ */
+export function normalizeRequest(
+    request: http.IncomingMessage,
+    validHosts: string[],
+) {
+    /* eslint-disable no-param-reassign */
+    const parsedUrl = url.parse(request.url!, true);
+    // @ts-expect-error
+    request.query = parsedUrl.query;
+    // TODO: make the namespace come from a config variable.
```
Comment (nicolas2bert) [inline lib/s3routes/routesUtils.ts:1020]: It is a good practice to have TODO attached with a JIRA ticket number to track todos :)

**Review by nicolas2bert — APPROVED** [review]
Looking good :)

File: lib/s3routes/routesUtils.ts (hunk 2 above)
Comment (jbertran) [reply lib/s3routes/routesUtils.ts:1020]: This one comes from existing code :grimacing:

File: lib/s3routes/routes/routePUT.ts (hunk 1 above)
Comment (jbertran) [reply lib/s3routes/routes/routePUT.ts]: Perhaps we can blanket-add ARSN-174 references to these TODOs. These all to track the second pass required to understand the business logic to type these accurately.

**Pull request merged**